docs/source/clickhouse/schema_design.rst

```diff
@@ -11,36 +21,64 @@ and not load them for every row for every query is part of the performance advan
 ClickHouse provides over a traditional RDBMS (like PostgreSQL).
 
 Commonly, a data schema contains a flexible key:value pair mapping
-(canonically: ``tags`` or ``contexts``) and stores that
+(canonically at Sentry: ``tags`` or ``contexts``) and stores that
 data in a column that contains two ``Nested`` arrays where the first array contains the keys
-of the dictionary and the second array contains the values. This works well when
-your dataset design gives you the ability to filter based on other attributes to a small
+of the dictionary and the second array contains the values, indexed by a bloom filter. This works well when
+your dataset and query design gives you the ability to filter for exact matches to a small
```

Comment:
This is not accurate. 

> stores that data in a column that contains two ``Nested`` arrays where the first array contains the keys of the dictionary and the second array contains the values

The column itself is `Nested`. The content is two standard arrays, one for keys and one for values.

> indexed by a bloom filter

We do not index the Nested column with a bloom filter. You can refer to the errors table schema for the details.
We index the tag keys array and a second column which is the tags hashmap. The tags hashmap is a simple array of integer where each entry is the hash of the `key=value` string.

Reply:
We also only do that on the tag values, we don't have the same indexes for contexts. So I don't know if this indexing pattern is "common".

Reply:
> We also only do that on the tag values, we don't have the same indexes for contexts. So I don't know if this indexing pattern is "common".

There's nothing special about contexts that makes them less index-able, right? I can make a note that we don't currently have bloom filter indexes on contexts but that doesn't seem inherent to the text.

Reply:
Also, I think I meant to imply that it's common across datasets, more than specific columns

Reply:
ok, adding a note about that